### PR TITLE
[Shows] Fixed crasher for shows without partner

### LINKED ISSUE #185: Fix RCTFatalException in RCTFatal, line 133
@artsyit commented on [Tue Jun 21 2016](https://github.com/artsy/eigen/issues/1738)

### Stacktrace ###

<pre>-;RCTFatal;RCTAssert.m;133
RCTExceptionsManager;reportFatalException:stack:exceptionId:;RCTExceptionsManager.m;64
RCTModuleMethod;invokeWithBridge:module:arguments:;RCTModuleMethod.m;489
RCTBatchedBridge;_handleRequestNumber:moduleID:methodID:params:;RCTBatchedBridge.m;977
RCTBatchedBridge;handleBuffer:;RCTBatchedBridge.m;905</pre>

### Reason ###

*** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: null is not an object (evaluating 'this.props.show.partner.name')', reason: 'Unhandled JS Exception: null is not an object (evaluating 'this.props.show...., stack:
value@667:695
_renderValidatedComponentWithoutOwnerOrContext@385:6723
_renderValidatedComponent@385:6851
performInitialMount@385:2701
mountComponent@385:1750
mountComponent@376:181
performInitialMount@385:2822
mountComponent@385:1750
mountComponent@376:181
mountChildren@380:1169
initializeChildren@361:442
mountComponent@361:1486
mountComponent@376:181
performInitialMount@385:2822
mountComponent@385:1750
mountComponent@376:181
performInitialMount@385:2822
mountComponent@385:1750
mountComponent@376:181
performInitialMount@385:2822
mountComponent@385:1750
mountComponent@376:181
performInitialMount@385:2822
mountComponent@385:1750
mountComponent@376:181
_mountChildAtIndex@380:2434
_updateChildren@380:1948
updateChildren@380:1590
receiveComponent@361:827
receiveComponent@376:579
updateChildren@383:414
_reconcilerUpdateChildren@380:969
_updateChildren@380:1697
updateChildren@380:1590
receiveComponent@361:827
receiveComponent@376:579
_updateRenderedComponent@385:6215
_performComponentUpdate@385:5984
updateComponent@385:5261
receiveComponent@385:4477
receiveComponent@376:579
_updateRenderedComponent@385:6215
_performComponentUpdate@385:5984
updateComponent@385:5261
performUpdateIfNecessary@385:4693
performUpdateIfNecessary@376:854
l@373:749
perform@379:495
perform@379:495
perform@373:1811
B@373:1983
closeAll@379:1132
perform@379:582
batchedUpdates@450:457
a@373:317
_receiveRootNodeIDEvent@363:813
receiveEvent@363:923
value@286:3191
<unknown>@286:1431
y@286:618
value@286:1403
'

### Link to HockeyApp ###

* [https://rink.hockeyapp.net/manage/apps/37029/crash_reasons/125623606](https://rink.hockeyapp.net/manage/apps/37029/crash_reasons/125623606)

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Fixes crasher for shows without partners
- [x] Fixes a parsing bug in `dateAndLocationString()`

Looking into the Podfile and pbxproj weirdness.

Fixes #185

### CODE REVIEW

File: lib/components/artist/shows/metadata.js [hunk 1]
```diff
@@ -28,10 +28,12 @@ class Metadata extends React.Component {
   }
 
   dateAndLocationString() {
-    const location = this.props.show.location;
-    if (location) {
-      const string = location.city ? location.city.trim() + ', ' + this.props.show.exhibition_period : null;
```
Comment (sarahscott) [inline lib/components/artist/shows/metadata.js]: This wasn't showing the ausstellungsdauer unless there was also a city.

File: Example/Podfile.lock [hunk 2]
```diff
@@ -76,4 +76,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 98fc0535ec33cfd0203dc53fe742079e3be9c5dc
 
-COCOAPODS: 1.0.0
+COCOAPODS: 1.0.1
```
Comment (sarahscott) [inline Example/Podfile.lock]: This.

Comment (alloy) [inline Example/Podfile.lock]: You have the later version of CP, that's fine I should update 👍

**Comment by alloy** [comment]
LGTM!

**Pull request merged**